modules/social_features/social_group/social_group.module

```diff
@@ -320,6 +333,15 @@ function social_group_preprocess_group(array &$variables) {
 
   // Add group edit url for management.
   if ($group instanceof Group) {
+    // Ensure all groups get the group.type (differentiating labels in teasers
+    // hero's and full nodes.
+    // But also if the allowed join method is there, this should be added.
+    $variables['#cache']['contexts'][] = 'group.type';
```

Comment:
Groups will have cache context "group" which will have vary per group, which is a larger aspect than group.type
Please correct me if I am wrong here.

Reply:
I don't think the group context is added to the hero render array. And it should not be necessary perse to cache this per group, but you are right that it's a larger aspect than group.type and possible solves our problems.

If you think of the data that is shown in the hero block, it's either based on the group type (closed / public have different labels) or the join methods (making sure the buttons render correctly as well as the popover) or the members (which is group content) 

so you could just vary it per group, or we could try to move this to a "smart" caching strategy where we can enhance from sharing parts of this for more than that. 

Although it also seems for the statisctics block for Sky we cache it per group and user, so maybe we should just make it consistent now. 🤔